In [1]:
import pandas as pd

In [95]:
data_2019 = pd.read_csv('raw_data/dcc-2019-pedestrian-footfall-count-jan-dec_14082020.csv')

In [75]:
data_2019.shape

(35036, 24)

In [103]:
data_2019.isna().sum()

Time                                                   0
O'Connell St Outside Pennys                            0
O'Connell St Outside Clerys                        26277
Mary Street                                            0
Capel Street                                           0
Aston Quay                                          5664
Grafton Street @ CompuB                             5472
Talbot Street North                                 5641
Doilier Street, Burgh Quay                          5484
Dawson Street Replacement                          31673
Dame Street                                         5472
Talbot Street South                                 5496
O'Connell St, Parnell St @ AIB                      7400
Grafton Street / Nassau Street / Suffolk Street     5472
College Green, Bank Of Ireland                      5472
Henry Street                                        6624
Westmoreland Street East                           26277
Dawson Street                  

In [106]:
data_2019_cleaned = data_2019.loc[:,['O\'Connell St Outside Clerys', 'Dawson Street Replacement', 'Westmoreland Street East', 'Dawson Street',\
'Liffey Street', 'Westmoreland Street West', 'Grafton Street']]

In [117]:
data_2019_cleaned['Dawson Street'].isnull().sum()

28269

In [111]:
data_2019_cleaned['Dawson Street'].tail(40)

34996   NaN
34997   NaN
34998   NaN
34999   NaN
35000   NaN
35001   NaN
35002   NaN
35003   NaN
35004   NaN
35005   NaN
35006   NaN
35007   NaN
35008   NaN
35009   NaN
35010   NaN
35011   NaN
35012   NaN
35013   NaN
35014   NaN
35015   NaN
35016   NaN
35017   NaN
35018   NaN
35019   NaN
35020   NaN
35021   NaN
35022   NaN
35023   NaN
35024   NaN
35025   NaN
35026   NaN
35027   NaN
35028   NaN
35029   NaN
35030   NaN
35031   NaN
35032   NaN
35033   NaN
35034   NaN
35035   NaN
Name: Dawson Street, dtype: float64

In [120]:
data_2019_fw = data_2019_cleaned.loc[:,['Dawson Street', 'Dawson Street Replacement']]

In [124]:
data_2019_fw = data_2019_fw.dropna()

In [125]:
data_2019_fw

Empty DataFrame
Columns: [Dawson Street, Dawson Street Replacement]
Index: []

In [79]:
len(data_2019['O\'Connell St Outside Clerys']) - data_2019['O\'Connell St Outside Clerys'].isna().sum()

8759

In [14]:
data_2020 = pd.read_csv('raw_data/jan-dec-2020-ped-data.csv')

In [89]:
data_2020 = data_2020.reindex(sorted(data_2020.columns), axis=1)

In [92]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8784 entries, 0 to 8783
Data columns (total 23 columns):
 #   Column                                           Non-Null Count  Dtype  
---  ------                                           --------------  -----  
 0   Aston Quay                                       8783 non-null   float64
 1   Bachelors Walk                                   8684 non-null   float64
 2   Capel Street - Removed from site 20/10           6791 non-null   float64
 3   College Green - Dame St Side                     8783 non-null   float64
 4   College Green @ Church Lane                      8783 non-null   float64
 5   College Green, Bank Of Ireland                   8783 non-null   float64
 6   Dame Street (Counter Missing)                    5897 non-null   float64
 7   Date & Time                                      8784 non-null   object 
 8   Dawson Street Replacement                        8783 non-null   float64
 9   Doilier Street, Burgh Quay    

In [23]:
data_2021 = pd.read_csv('raw_data/dcc-2021-pedestrian-footfall-31072021.csv')

In [90]:
data_2021 = data_2021.reindex(sorted(data_2021.columns), axis=1)

In [93]:
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5088 entries, 0 to 5087
Data columns (total 35 columns):
 #   Column                                           Non-Null Count  Dtype  
---  ------                                           --------------  -----  
 0   Aston Quay/Fitzgeralds                           2796 non-null   float64
 1   Bachelors walk/Bachelors way                     5087 non-null   float64
 2   Baggot st lower/Wilton tce inbound               5079 non-null   float64
 3   Baggot st upper/Mespil rd/Bank                   5079 non-null   float64
 4   Capel st/Mary street                             816 non-null    float64
 5   College Green/Bank Of Ireland                    5060 non-null   float64
 6   College Green/Church Lane                        5087 non-null   float64
 7   College st/Westmoreland st                       5087 non-null   float64
 8   D'olier st/Burgh Quay                            5087 non-null   float64
 9   Dame Street/Londis            

# Liste des capteurs ajoutés en 2021

2   Baggot st lower/Wilton tce inbound               5079 non-null   float64
 3   Baggot st upper/Mespil rd/Bank 

 15  Grand Canal st upp/Clanwilliam place             5078 non-null   float64
 16  Grand Canal st upp/Clanwilliam place/Google

Newcomen Bridge/Charleville mall inbound         5075 non-null   float64
 21  Newcomen Bridge/Charleville mall outbound        5074 non-null   float64
 22  North Wall Quay/Samuel Beckett bridge East       5087 non-null   float64
 23  North Wall Quay/Samuel Beckett bridge Wes

27  Phibsborough Rd/Enniskerry Road                  5087 non-null   float64
 28  Phibsborough Rd/Munster St                       798 non-null    float64
 29  Richmond st south/Portabello Harbour inbound     5079 non-null   float64
 30  Richmond st south/Portabello Harbour outbound 